File: rain_forest_regression.py
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

dataset = pd.read_csv('/Users/sparshgarg/Documents/SantaClara/Machine Learning/Machine Learning_Udemy/Part 2 - Regression/Random_Forest_Regression/Position_Salaries.csv')
X = dataset.iloc[:,1:2].values
Y = dataset.iloc[:, 2].values

from sklearn.ensemble import RandomForestRegressor
regressor = RandomForestRegressor(n_estimators=300, random_state=0)
regressor.fit(X,Y)
y_pred = regressor.predict('6.5')
print(y_pred)
x_grid = np.arange(min(X),max(X),0.01)
x_grid = x_grid.reshape(len(x_grid),1)

plt.scatter(X,Y)
plt.plot(x_grid,regressor.predict(x_grid))
plt.show()

File: rain_forest_regression.R
dataset = read.csv('/Users/sparshgarg/Documents/SantaClara/Machine Learning/Machine Learning_Udemy/Part 2 - Regression/Random_Forest_Regression/Position_Salaries.csv')
dataset = dataset[2:3]

library(randomForest)
set.seed(1234)
regressor = randomForest(x = dataset[1], y = dataset$Salary, ntree = 200)
y_pred = predict(regressor,data.frame(Level = 6.5))

x_grid = seq(min(dataset$Level), max(dataset$Level), 0.01)
library(ggplot2)
ggplot()+geom_point(aes(dataset$Level, dataset$Salary), colour = 'red')+
  geom_line(aes(x_grid, predict(regressor, newdata = data.frame(Level = x_grid)) ), colour = 'blue')
